nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-cluster-protocol/src/main/java/org/apache/nifi/cluster/coordination/heartbeat/HeartbeatMonitor.java

```diff
@@ -59,6 +59,13 @@
     void purgeHeartbeats();
 
     /**
+     * Returns when the heartbeats were purged last.
+     *
+     * @return when the heartbeats were purged last
+     */
+    long getPurgeTimestamp();
```

Comment:
This feels a bit odd to add to HeartbeatMonitor. I think this is a function of the specific implementation of AbstractHeartbeatMonitor. I think this should instead be an abstract method on the AbstractHeartbeatMonitor instead of going into this interface.